packages/stable/src/api/documents/documentsApi.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2022 Cognite AS
+
+import { BaseResourceAPI, CDFHttpClient, MetadataMap } from '@cognite/sdk-core';
+
+import {
+  Document,
+  DocumentSearchResponse,
+  DocumentSearchRequest,
+} from '../../types';
+
+export class DocumentsAPI extends BaseResourceAPI<Document> {
+  constructor(...args: [string, CDFHttpClient, MetadataMap]) {
+    super(...args);
+  }
+
+  /**
+   * [Search for documents](https://docs.cognite.com/api/v1/#operation/documentsSearch)
+   *
+   * ```js
+   * const documents = await client.documents.search({
+   *   search: {
+   *     query: 'Stuck pipe'
+   *   },
+   *   filter: {
+   *     in: {
+   *       property: ['type'],
+   *       values: ['Document', 'PDF']
+   *     }
+   *   },
+   *   limit: 5
+   * });
+   * ```
+   */
+  public search = (
+    query: DocumentSearchRequest
+  ): Promise<DocumentSearchResponse> => {
+    return this.searchDocuments<DocumentSearchResponse>(query);
+  };
+
+  private async searchDocuments<ResponseType>(
```

Comment:
You have to add:

`return this.addToMapAndReturn(response.data, response);`

This is so users can get headers or other data. It's built into all the helper methods of the library, but when you write custom methods you need to specify this manually.

Reply:
Oh, maybe I could instead use one of those helper methods that have it built in? I'll look into that

Reply:
Maybe you can use `callSearchEndpoint`.

Reply:
I ended up following your first suggestion, which seems to work fine